src/compiler/checker.ts

```diff
@@ -29188,10 +29188,6 @@ namespace ts {
                 }
                 checkGrammarForDisallowedTrailingComma(elements, Diagnostics.A_rest_parameter_or_binding_pattern_may_not_have_a_trailing_comma);
 
-                if (node.name.kind === SyntaxKind.ArrayBindingPattern || node.name.kind === SyntaxKind.ObjectBindingPattern) {

```

Comment:
note that this never checked BindingRestProperty because that has a parse error: https://agentcooper.github.io/typescript-play/#code/DYUwLgBA3gdHUF8EQLzQQGmnGBtAusmogNxA